cmd/manager/main.go

```diff
@@ -154,7 +156,19 @@ func main() {
 
 	mux.HandleFunc("/healthz", func(w http.ResponseWriter, _ *http.Request) {
 		w.WriteHeader(http.StatusOK)
-		w.Write([]byte("ok\n"))
+		w.Write([]byte("ok"))
+	})
+
+	mux.HandleFunc("/readyz", func(w http.ResponseWriter, _ *http.Request) {
+		namespaces := corev1.NamespaceList{}
+		err = mgr.GetClient().List(ctx, &client.ListOptions{}, &namespaces)
```

Comment:
Why do we check namespaces in particular? Could you provide a comment with an explanation in the code of this func? I'm not (yet? 😜) trying to argue with this choice, rather trying to understand it in the first place.

Reply:
I'm wondering if instead of list namespaces we should list GitOpsConfigs. What do you think?

Reply:
Generally I do not see benefit of listing GitOpsConfigs over Namespaces. Virtually it can be anything (e.g.: ConfigMaps, Pods, etc.) because we are only interested here in if there is error while calling API or not. We do not even care if there actually was something returned, but if we were, Namespace is still the most obvious choice, because there would always be some regardless of level of access Pod has. We could add here check if returned list is not empty and that would make some sense. But let's remember that this operation is frequent, and not cached via informer mechanism (which is essential for this check). Also I didn't want to produce heavy load of code there. 

Let me know what you think :)

Reply:
@mkyc thanks for the comment with explanation in code, now I understand and it makes sense to me, with my limited knowledge of Kubernetes.

Reply:
We can try using namespace and see how well it works.